packages/x-charts/src/ChartsLabel/ChartsLabelGradient.tsx

```diff
@@ -43,32 +44,40 @@ export interface ChartsLabelGradientProps {
   sx?: SxProps<Theme>;
 }
 
+const applyRtl = (isRtl: boolean | undefined, value: number) => (isRtl ? value - 180 : value);
+
 const getRotation = (
   direction?: 'vertical' | 'horizontal',
   reverse?: boolean,
   rotate?: boolean,
+  isRtl?: boolean,
 ) => {
   if (!rotate && reverse) {
-    return direction === 'vertical' ? 90 : 180;
+    return direction === 'vertical' ? 90 : applyRtl(isRtl, 180);
   }
 
   if (rotate && !reverse) {
-    return direction === 'vertical' ? 0 : 90;
+    return direction === 'vertical' ? applyRtl(isRtl, 0) : 90;
   }
 
   if (rotate && reverse) {
-    return direction === 'vertical' ? 180 : -90;
+    return direction === 'vertical' ? applyRtl(isRtl, 180) : -90;
   }
 
-  return direction === 'vertical' ? -90 : 0;
+  return direction === 'vertical' ? -90 : applyRtl(isRtl, 0);
 };
```

Comment:
It's a bit hard to read all the cases. What about

```js
let angle = (direction === 'vertical' ? -90: 0) + (rotate ? 90 : 0) + (reverse ? 180 : 0);

if(isRtl && Math.abs(angle) % 180 === 90){ // dir='rtl' and horizontal gradient
  return angle + 180;
}
return angle;
```

Reply:
Or another more readable approach to know if you need to add 180° for RtL reverse could be

```js
const shouldRtlReverse = direction === 'vertical' && rotate || direction === 'horizontal' && !rotate
```

Reply:
I fear it is not as simple 😅 

The angle is not a reliable way to say if the gradient is actually horizontal or not due to the rotate flag 😓 
I left it as `if(isRtl && direction === 'horizontal')` and seems to work on all cases

![Screenshot 2025-01-10 at 14 20 06](https://github.com/user-attachments/assets/89ecec41-1851-4073-a8a2-ff53f8090ed9)